main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -481,6 +483,87 @@ object CompletionProvider {
     }.toList
   }
 
+  /**
+    * Format type params in the right form to be inserted as a snippet
+    */
+  private def formatTParamsSnippet(tparams: List[TypedAst.TypeParam]): String = {
+    tparams match {
+      case Nil => ""
+      case _ => tparams.zipWithIndex.map {
+                  case (tparam, idx) => "$" + s"{${idx + 1}:${tparam.name}}"
+                }.mkString("[", ", ", "]")
+    }
+  }
+
+  /**
+    * Format type params in the right form to be displayed in the list of completions
+    */
+  private def formatTParams(tparams: List[TypedAst.TypeParam]): String = {
+    tparams match {
+      case Nil => ""
+      case _ => tparams.map(_.name).mkString("[", ", ", "]")
+    }
+  }
+
+  val simpleTypeNames = List(
+    "Unit",
+    "Null",
+    "Bool",
+    "Char",
+    "Float32",
+    "Float64",
+    "Int8",
+    "Int16",
+    "Int32",
+    "Int64",
+    "BigInt",
+    "String",
+    "Array",
+    "Ref",
+    "Channel",
+    "Lazy"
```

Comment:
We also have types like tuples and records, but I guess we cannot easily complete those?

Reply:
I can't think of any easy way to do so, no, but please do let me know if you can think of one!